Sources/NIOSSL/TLSConfiguration.swift

```diff
@@ -212,6 +212,18 @@ public struct TLSConfiguration {
     /// Whether renegotiation is supported.
     public var renegotiationSupport: NIORenegotiationSupport
 
+    /// Whether to trust the commonName field when validating certificate hostnames.
+    ///
+    /// The commonName field of certificates has been deprecated for use as a validation source
+    /// for nearly two decades. As browsers and the WebPKI now forbid the usage of common names
+    /// in certificates, swift-nio-ssl defaults to this strict level of enforcement as well.
+    ///
+    /// However, some older infrastructure does not issue certificates with valid subject alternative
+    /// name fields. For this reason, we currently allow an override for trusting the common name. This
+    /// is a temporary measure to allow users to work around these limitations, but we will remove this
+    /// functionality in future.
+    public var trustCommonName: Bool
```

Comment:
Should we (assuming it's possible) mark the setter for this as deprecated right from the start?